add dicom reading

HGeerlings opened the pull request:
Add basic dicom reading. Must generalize use tomorrow.

HGeerlings committed:
add dicom reading

Add basic dicom reading. Must generalize use tomorrow.

bkappes commented:
Don't commit this quite yet. I'll make a few changes and add them to the pull request when I'm working on my computer; a device more suited to code editing than an iPhone!

bkappes commented on scratch/PCA/extract_data.ipynb:
```diff
@@ -18,40 +18,81 @@
     "\"\"\"\n",
     "\n",
     "from __future__ import division\n",
```
Not the best place for this note, but I can't comment on the file as a whole. Parsing out ipython notebooks isn't really doable in a pull request -- the JSON formatting makes it to difficult to read. We can go through this as a demo. Do you have 30 minutes later today, say at 11:00? I'll be in my office, if I've managed to get Yasuhito going on the XCT, otherwise I'll be in W230.

HGeerlings replied:
Sure Branden, 

I can meet at 11:15, would that work for you?

bkappes replied:
Yes. I’m in my office.

> On Aug 16, 2016, at 11:01 AM, HGeerlings notifications@github.com wrote:
> 
> In scratch/PCA/extract_data.ipynb https://github.com/csm-adapt/tracr/pull/4#discussion_r74976256:
> 
> > @@ -18,40 +18,81 @@
> >      "\"\"\"\n",
> >      "\n",
> >      "from **future** import division\n",
> > Sure Branden,
> 
> I can meet at 11:15, would that work for you?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub https://github.com/csm-adapt/tracr/pull/4/files/c34dbfd1457b395e51fb01c9960400ba80f95fbc#r74976256, or mute the thread https://github.com/notifications/unsubscribe-auth/AFjEmRkrgnsPPTOpTYPOy43qCZgect5pks5qgez7gaJpZM4Jk3y6.

HGeerlings replied:
## Branden, can you meet briefly?

Henry

> On Aug 16, 2016, at 11:05 AM, bkappes notifications@github.com wrote:
> 
> In scratch/PCA/extract_data.ipynb https://github.com/csm-adapt/tracr/pull/4#discussion_r74976937:
> 
> > @@ -18,40 +18,81 @@
> >      "\"\"\"\n",
> >      "\n",
> >      "from **future** import division\n",
> > Yes. I’m in my office.
> > … x-msg://5/#
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> 
>  https://github.com/csm-adapt/tracr/pull/4#discussion_r74976256 https://github.com/csm-adapt/tracr/pull/4/files/c34dbfd1457b395e51fb01c9960400ba80f95fbc#r74976256 https://github.com/notifications/unsubscribe-auth/AFjEmRkrgnsPPTOpTYPOy43qCZgect5pks5qgez7gaJpZM4Jk3y6
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub https://github.com/csm-adapt/tracr/pull/4/files/c34dbfd1457b395e51fb01c9960400ba80f95fbc#r74976937, or mute the thread https://github.com/notifications/unsubscribe-auth/AMZvXbjt61Nz-r-5qKqcwrHmzvwTlp4cks5qge3ogaJpZM4Jk3y6.

bkappes replied:
Sorry, Henry, but I'm already gone. We can Skype at 6:30 if you like. I'll be in Chicago for the remainder of the week.

Branden

Sent from my iPhone

> On Aug 16, 2016, at 5:06 PM, HGeerlings notifications@github.com wrote:
> 
> In scratch/PCA/extract_data.ipynb:
> 
> > @@ -18,40 +18,81 @@
> >      "\"\"\"\n",
> >      "\n",
> >      "from **future** import division\n",
> > Branden, can you meet briefly? -- Henry
> > …
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub, or mute the thread.

bkappes commented on tracr/io/dicom/read.py:
```diff
@@ -0,0 +1,35 @@
+
+def read(dcm_files):
+	# Import image and details. Input is a list of filename strings
+	layer = dicom.read_file(dcm_files[0])
```
What happens if the user passes only a single dicom file?

bkappes replied:
What happens if the file doesn't exist?

bkappes commented on tracr/io/dicom/read.py:
```diff
@@ -0,0 +1,35 @@
+def read(dcm_files):
+	# Import image and details. Input is a list of filename strings
+	layer = dicom.read_file(dcm_files[0])
+	dicom_array = np.zeros((int(layer.Rows), int(layer.Columns), len(dcm_files)))
```
What happens if not all the frames are the same size? Is this a case we should handle? If so, how?

bkappes commented on tracr/io/dicom/read.py:
```diff
@@ -0,0 +1,35 @@
+	# Import image and details. Input is a list of filename strings
+	layer = dicom.read_file(dcm_files[0])
+	dicom_array = np.zeros((int(layer.Rows), int(layer.Columns), len(dcm_files)))
+	pixel_size = float(layer.PixelSpacing[0])*1000
```
Indexing PixelSpacing is curious. Does it store both x and y pixel dimensions, but you tacitly assume they will always be the same?

HGeerlings committed:
ignore ipynb, also change dicom.read

Added _regularize functionality (still WIP), and also consider single
file usage.
